themes/src/main/resources/theme/base/login/resources/js/webauthnAuthenticate.js

```diff
@@ -0,0 +1,78 @@
+import { base64url } from "../../../common/keycloak/node_modules/rfc4648/lib/rfc4648.js";
+
+export function authenticateByWebAuthn(input) {
+    if (!input.isUserIdentified) {
+        doAuthenticate([], input.challenge, input.userVerification, input.rpId, input.createTimeout, input.errmsg)
+            .then(result => returnSuccess(result))
+            .catch(err => returnFailure(err));
+        return;
+    }
+    checkAllowCredentials(input.challenge, input.userVerification, input.rpId, input.createTimeout, input.errmsg);
+}
+
+function checkAllowCredentials(challenge, userVerification, rpId, createTimeout, errmsg) {
+    const allowCredentials = [];
+    const authn_use = document.forms['authn_select'].authn_use_chk;
+    if (authn_use !== undefined) {
+        if (authn_use.length === undefined) {
+            allowCredentials.push({
+                id: base64url.parse(authn_use.value, {loose: true}),
+                type: 'public-key',
+            });
+        } else {
+            for (const entry of authn_use) {
+                allowCredentials.push({
+                    id: base64url.parse(entry.value, {loose: true}),
+                    type: 'public-key',
+                });
+            }
+        }
+    }
+    doAuthenticate(allowCredentials, challenge, userVerification, rpId, createTimeout, errmsg)
+        .then(result => returnSuccess(result))
+        .catch(err => returnFailure(err));
+}
```

Comment:
Let's use `async`/`await` here as well.
```suggestion
async function checkAllowCredentials(challenge, userVerification, rpId, createTimeout, errmsg) {
    const allowCredentials = [];
    const authn_use = document.forms['authn_select'].authn_use_chk;
    if (authn_use !== undefined) {
        if (authn_use.length === undefined) {
            allowCredentials.push({
                id: base64url.parse(authn_use.value, {loose: true}),
                type: 'public-key',
            });
        } else {
            for (const entry of authn_use) {
                allowCredentials.push({
                    id: base64url.parse(entry.value, {loose: true}),
                    type: 'public-key',
                });
            }
        }
    }
    
    try {
        const result = await doAuthenticate(allowCredentials, challenge, userVerification, rpId, createTimeout, errmsg);
        returnSuccess(result
    } catch (error) {
        returnFailure(error);
    }
}
```

Reply:
I fixed the codes as you suggested.